static/src/javascripts-legacy/projects/commercial/modules/dfp/prepare-switch-tag.js

```diff
@@ -74,19 +74,25 @@ define([
     function pushAdUnit(dfpDivId, sizeMapping) {
 
         var __switch_zero = window.__switch_zero;
+        var promises = [];
 
         if (__switch_zero) {
             var adUnitIds = findAdUnitIds(sizeMapping.size);
 
             adUnitIds.forEach(function(adUnitId) {
```

Comment:
Just an idea here: you could return a `map` of `adUnitId`s to `Promise`s objects and just return `Promise.resolve()` in the other case.